desc/objectives/_profiles.py

```diff
@@ -0,0 +1,549 @@
+        if self._grid is None:
+            grid = LinearGrid(
+                L=eq.L_grid,
+                M=eq.M_grid,
+                N=eq.N_grid,
+                NFP=eq.NFP,
+                sym=eq.sym,
+                axis=False,
+            )
+        else:
+            grid = self._grid
+
+        if isinstance(self._target, Profile):
+            self._target = self._target(grid.nodes[grid.unique_rho_idx])
+
+        self._dim_f = grid.num_rho
+        self._data_keys = ["rho", "iota", "iota_r", "iota_rr"]
+        self._args = get_params(
+            self._data_keys,
+            obj="desc.equilibrium.equilibrium.Equilibrium",
+            has_axis=grid.axis.size,
+        )
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        self._profiles = get_profiles(self._data_keys, obj=eq, grid=grid)
+        self._transforms = get_transforms(self._data_keys, obj=eq, grid=grid)
+        self._constants = {
+            "transforms": self._transforms,
+            "profiles": self._profiles,
+        }
+
+        timer.stop("Precomputing transforms")
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        self._eps = 1e2 * jnp.finfo(jnp.array([1.0]).dtype).eps
+
+        super().build(eq=eq, use_jit=use_jit, verbose=verbose)
+
+    def compute(self, *args, **kwargs):
+        """Compute shear profile errors.
+
+        Parameters
+        ----------
+        R_lmn : ndarray
+            Spectral coefficients of R(rho,theta,zeta) -- flux surface R coordinate (m).
+        Z_lmn : ndarray
+            Spectral coefficients of Z(rho,theta,zeta) -- flux surface Z coordinate (m).
+        L_lmn : ndarray
+            Spectral coefficients of lambda(rho,theta,zeta) -- poloidal stream function.
+        i_l : ndarray
+            Spectral coefficients of iota(rho) -- rotational transform profile.
+        c_l : ndarray
+            Spectral coefficients of I(rho) -- toroidal current profile.
+        Psi : float
+            Total toroidal magnetic flux within the last closed flux surface (Wb).
+
+        Returns
+        -------
+        shear : ndarray
+            Normalized radial derivative of the rotational transform.
+
+        """
+        params, constants = self._parse_args(*args, **kwargs)
+        if constants is None:
+            constants = self.constants
+        data = compute_fun(
+            "desc.equilibrium.equilibrium.Equilibrium",
+            self._data_keys,
+            params=params,
+            transforms=constants["transforms"],
+            profiles=constants["profiles"],
+        )
+        shear = jnp.where(
+            jnp.abs(data["iota"]) < self._eps,
+            1 + data["rho"] * data["iota_rr"] / data["iota_r"],
+            data["rho"] * data["iota_r"] / data["iota"],
+        )
+        shear = jnp.where(jnp.all(jnp.abs(data["iota"])) < self._eps, 0, shear)
```

Comment:
This might cause NaN in the gradient if the previous `jnp.where` divided by zero and then it gets overwritten. Might be better to use `cond` to first check if iota is all zero, then use jnp.where inside that to handle the case where its zero at a point.